src/modules/compliance/src/lib/Evaluator.cpp

```diff
@@ -16,241 +18,563 @@
+    return result;
+}
+
+void MmiFormatter::FormatNode(const IndicatorsTree::Node& node, std::ostringstream& result) const
+{
+    if (node.procedureName == "anyOf" || node.procedureName == "allOf")
+    {
+        bool first = true;
+        result << "{ " << node.procedureName << ": [";
+        for (const auto& child : node.children)
+        {
+            if (!first)
             {
-                OsConfigLogInfo(log, "No remediation function found for '%s', using audit function", name);
-                fn = procedure->second.first;
+                result << ", ";
             }
+            FormatNode(*child.get(), result);
+            first = false;
+        }
+        result << "]} == ";
+        if (node.status == Status::Compliant)
+        {
+            result << "TRUE";
         }
         else
         {
-            fn = procedure->second.first;
+            result << "FALSE";
         }
-        if (nullptr == fn)
+    }
+    else if (node.procedureName == "not")
+    {
+        result << "{ " << node.procedureName << ": ";
+        FormatNode(*node.children[0].get(), result);
+        result << "} == ";
+        if (node.status == Status::Compliant)
         {
-            logstream << "ERROR: Function not found";
-            OsConfigLogError(log, "Function not found");
-            return Error("Function not found");
+            result << "FALSE";
         }
-        logstream << "{ " << name << ": ";
-        auto result = fn(arguments, *mContext);
-        logstream << " } == ";
-        if (!result.HasValue())
+        else
         {
-            logstream << "FAILURE";
-            return {result.Error()};
+            result << "TRUE";
         }
+    }
+    else
+    {
+        result << "{ " << node.procedureName << ": ";
 
-        if (result.Value() == true)
+        bool first = true;
+        for (const auto& indicator : node.indicators)
+        {
+            if (!first)
+            {
+                result << ", ";
+            }
+            result << indicator.message;
+            first = false;
+        }
+        result << " } == ";
+        if (node.status == Status::Compliant)
         {
-            logstream << "TRUE";
+            result << "TRUE";
         }
         else
         {
-            logstream << "FALSE";
+            result << "FALSE";
         }
+    }
+}
 
-        return result.Value() ? Status::Compliant : Status::NonCompliant;
+Result<std::string> MmiFormatter::Format(const IndicatorsTree& indicators) const
+{
+    std::ostringstream result;
+    const auto* node = indicators.GetRootNode();
+    assert(nullptr != node);
+    if (node->children.empty())
+    {
+        return Error("No children found", EINVAL);
     }
-    return Error("Unreachable"); // unreachable
+    node = node->children[0].get();
+    assert(nullptr != node);
+    FormatNode(*node, result);
+    return (indicators.GetRootNode()->status == Status::Compliant ? "PASS" : "") + result.str();
```

Comment:
>PASS [](http://example.com/codeflow?start=69&length=4)

Use the SECURITY_AUDIT_PASS macro defined in Reasons.h for this? Just in case we'll change this mark in the future.